packages/web/src/components/twitter-auth/TwitterAuthButton.tsx

```diff
@@ -0,0 +1,28 @@
+import { audiusBackendInstance } from 'services/audius-backend/audius-backend-instance'
+
+import TwitterButton, {
+  TwitterButtonProps
+} from '../twitter-button/TwitterButton'
+
+import TwitterAuth, { TwitterAuthProps } from './TwitterAuth'
+
+type TwitterAuthButtonProps = TwitterButtonProps &
+  Omit<TwitterAuthProps, 'requestTokenUrl' | 'loginUrl'>
+
+const TwitterAuthButton = (props: TwitterAuthButtonProps) => {
+  const { onClick, onFailure, onSuccess } = props
+  return (
+    <TwitterAuth
```

Comment:
Personally not in love with this auth wrapping button thing, could it all just be Twitter button?

Reply:
Totally agree, this is the legacy implementation. TikTok uses a hook. I could convert twitter and ig to hooks too but it really increases surface area